cli: v1.10.0 promotion step fails

# Checklist

* [x] I've searched the issue queue to verify this is not a duplicate bug report.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `kargo version`.
* [x] I've pasted logs, if applicable.

# Description

After upgraded to v1.10.0 on my client and server, I'm having issues of running the promotion step through the cli. The UI for promotion is still working well, just the cli is having issues.

# Steps to Reproduce

Just running `kargo promote --project=x --stage=x --freight="x"`

# Version

```
Client Version: v1.10.0
Server Version: v1.10.0
```

# Logs

```
kargo promote --project=x --stage=x --freight="x"                                                                    
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x101e407a0]

goroutine 1 [running]:
github.com/akuity/kargo/api/v1alpha1.(*Promotion).GetObjectKind(0x3780acb3d8a8?)
	<autogenerated>:1
k8s.io/cli-runtime/pkg/printers.(*TypeSetterPrinter).PrintObj(0x3780ac7fed60, {0x1067e5af0, 0x0}, {0x1067d83b8, 0x3780ac582028})
	/go/pkg/mod/k8s.io/cli-runtime@v0.34.3/pkg/printers/typesetter.go:45 +0x5c
github.com/akuity/kargo/pkg/cli/cmd/promote.(*promotionOptions).run(0x3780ac9d8500, {0x106802d50, 0x106a125a0})
	/__w/kargo/kargo/pkg/cli/cmd/promote/promote.go:228 +0x780
github.com/akuity/kargo/pkg/cli/cmd/promote.NewCommand.func1(0x3780ac9bd508, {0x3780ac9dc7b0?, 0x4?, 0x10266da39?})
	/__w/kargo/kargo/pkg/cli/cmd/promote/promote.go:93 +0x4c
github.com/spf13/cobra.(*Command).execute(0x3780ac9bd508, {0x3780ac9dc750, 0x3, 0x3})
	/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1015 +0x814
github.com/spf13/cobra.(*Command).ExecuteC(0x3780ac66c908)
	/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1071
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1064
main.main()
	/__w/kargo/kargo/cmd/cli/main.go:31 +0x20c
```

--- a/pkg/cli/cmd/promote/promote.go
+++ b/pkg/cli/cmd/promote/promote.go
@@ -212,14 +212,10 @@ func (o *promotionOptions) run(ctx context.Context) error {
 		if err != nil {
 			return fmt.Errorf("marshal promotion: %w", err)
 		}
-		// The response is {"promotion": {...}}
-		var result struct {
-			Promotion *kargoapi.Promotion `json:"promotion"`
-		}
-		if err = json.Unmarshal(promoJSON, &result); err != nil {
+		promo := &kargoapi.Promotion{}
+		if err = json.Unmarshal(promoJSON, promo); err != nil {
 			return fmt.Errorf("unmarshal promotion: %w", err)
 		}
-		promo := result.Promotion
 		if o.Wait {
 			if err = o.waitForPromotion(ctx, nil, promo); err != nil {
 				return fmt.Errorf("wait for promotion: %w", err)

--- a/pkg/server/promote_to_stage_v1alpha1.go
+++ b/pkg/server/promote_to_stage_v1alpha1.go
@@ -284,5 +284,5 @@ func (s *server) promoteToStage(c *gin.Context) {
 		s.recordPromotionCreatedEvent(ctx, promotion, freight)
 	}
 
-	c.JSON(http.StatusCreated, gin.H{"promotion": promotion})
+	c.JSON(http.StatusCreated, promotion)
 }

